### ISSUE TITLE
Some links are missing a :hover style

### ISSUE THREAD

**Issue by carolinan** [WordPress/twentytwenty#442]
Links need to have a visual change when hovered.

Most of the links that are missing a hover style are in the content,
where the link is always underlined:
Links in post and page content
Links in captions, both in blocks and normal.
Blocks examples:
Links in cover block text (both image and video)
Links in quote blocks
Links in media & text block

_Edge case:_
Linked images in post content is missing any indication except the mouse pointer change.

----
But then we have some links in blocks that are not underlined in their default state, 
but underlined when hovered. The opposite of the other links in the content.

Archive and category widgets
Latest comments widget

Latest posts widget:
This one is different, the first item of the latest posts when in list view has _no underline,_
but the rest of the items have an underline when hovered.

----

Page breaks
Edit this
Category links

Comment author name
Links in comment content

Horizontal menu: 
current-menu-item is always underlined.

**Comment by nielslange** [comment]
Happy to work on this, @carolinan.

@andersnoren  What do you have in mind for styling links? Should we change the current solid underline to a dotted underline by default and use a solid underline on hover?

**Comment by acosmin** [comment]
the menu and search buttons (+ their text) are black for me in single view with cover template 👀 

submenus in desktop view have the accent color on links, I think it should be white?

**Comment by andersnoren** [comment]
@acosmin Some stuff still to work out with the element targeting :) I'll submit another PR in a bit.

@nielslange Awesome! For elements that have an underline to start with, removing the underline on hover will be fine. Elements that don't have an underline to start with get an underline on hover.
